### PR TITLE
Add kvutils ledger dumps and integrity checker

### PR DESCRIPTION
**Pull request by dajmaki**
kvutils now supports dumping the ledger to a file via environment
variable: "KVUTILS_LEDGER_DUMP=/tmp/ledger.dump".

The integrity checker tool allows re-processing of ledger dumps to
validate that all submissions, state and log entries can be parsed
and processed. This forms the basis of kvutils data continuity test
suite.

Currently the checker strictly enforces that re-processed
outputs match the inputs exactly, which for now guarantees forwards
and backwards compatibility. The checker will be adapted when
forwards-incompatible changes or new kvutils message versions arise.

A select collection of ledger dumps will be maintained in a separate
repository (to be decided) and these dumps are to be validated by
the DAML CI.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Debug.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+package com.daml.ledger.participant.state.kvutils
+
+import java.io.{DataOutputStream, FileOutputStream}
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+
+import scala.collection.JavaConverters._
+
+/** Utilities for debugging kvutils. */
+object Debug {
```
Comment (gerolf-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Debug.scala]: Are you 100% certain that kvutils won't be used multiple times in the same process (e.g. some test)?

File: 3rdparty/jvm/io/dropwizard/metrics/BUILD [hunk 2]
```diff
@@ -28,3 +28,17 @@ java_library(
 )
 
 
+
+java_library(
+    name = "metrics_jvm",
+    exports = [
+        "//3rdparty/jvm/org/slf4j:slf4j_api",
```
Comment (stefanobaghino-da) [inline 3rdparty/jvm/io/dropwizard/metrics/BUILD]: Is this supposed to be here?

File: ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import com.daml.ledger.participant.state.kvutils.{DamlKvutils => Proto, _}
+import com.daml.ledger.participant.state.v1.{Configuration, Update}
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml.lf.engine.Engine
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object Helpers {
+  def time[T](act: () => T): (Long, T) = {
```
Comment (stefanobaghino-da) [inline ledger/participant-state/kvutils/tools/src/main/scala/com/daml/ledger/participant/state/kvutils/tools/IntegrityCheck.scala]: Minor, but a by-name parameter is probably good enough here.
```suggestion
  def time[T](act: => T): (Long, T) = {
```

File: 3rdparty/jvm/io/dropwizard/metrics/BUILD (hunk 2 above)
Comment (dajmaki) [reply 3rdparty/jvm/io/dropwizard/metrics/BUILD]: Yes

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Debug.scala (hunk 1 above)
Comment (dajmaki) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Debug.scala]: 99%, and 100% for this functionality.

**Pull request merged**